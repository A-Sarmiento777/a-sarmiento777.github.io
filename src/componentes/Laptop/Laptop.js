import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";

const Laptop = ({ agregarProductoAlCarrito, agregarProductoAlCarritoFav, removeItemFromCartFav, cart, favProducts, laptop }) => {
    return (
        <div className="card mr-4 mb-4 p-3 shadow p-3 mb-5 bg-white rounded " style={{ width: "200px", minWidth: "30px" }}>
            <div className="h6">{laptop?.name}</div>
            <div className="h6"> <small> Price: {laptop?.price} kr </small></div>
            <div className="mx-auto">
                <img src={process.env.PUBLIC_URL + `/Assets/laptops/${laptop?.name}.jpg`} width='150' alt="logos" className="img-fluid py-2" />
            </div>
            <div>
                {favProducts?.includes(laptop?.name) ? <button className="bg-light border-0" onClick={() => removeItemFromCartFav(laptop.id + 'l')}>
                    <img src={process.env.PUBLIC_URL + `/Assets/heart.png`} width='20' alt="logo" />
                </button> : <button className="bg-light border-0" onClick={() => agregarProductoAlCarritoFav(laptop?.id + 'l', laptop?.name, laptop?.price, 'laptops')}>
                    <img src={process.env.PUBLIC_URL + `/Assets/fav.png`} width='20' alt="logo" />
                </button>}
                {cart?.includes(laptop?.name) ? <Link to='/checkout'><button className='bg-success px-1 rounded ml-2'>View Cart</button></Link> : <button className="bg-warning px-1 rounded ml-2" onClick={() => agregarProductoAlCarrito(laptop?.id + 'l', laptop?.name, laptop?.price, 'laptops', 1)}>Add</button>}
            </div>
        </div>
    );
};

export default Laptop;
