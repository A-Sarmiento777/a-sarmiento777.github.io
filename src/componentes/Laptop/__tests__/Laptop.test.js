// Link.react.test.js
import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import Laptop from "../Laptop";

test("Laptop Card test", () => {
  const agregarProductoAlCarrito = jest.fn(() => 1);

  let testRenderer = renderer.create(
    <Laptop
      agregarProductoAlCarrito={agregarProductoAlCarrito}
      agregarProductoAlCarritoFav={() => {}}
      removeItemFromCartFav={() => {}}
      cart={[]}
      favProducts={[]}
      laptop={{ id: 1, name: "Test", price: "6.7" }}
    ></Laptop>
  );

  let tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[3].children[1].props.onClick();

  expect(agregarProductoAlCarrito).toHaveBeenCalled();
  expect(agregarProductoAlCarrito).toBeCalledTimes(1);

  testRenderer.update(
    <Router>
      <Laptop
        agregarProductoAlCarrito={agregarProductoAlCarrito}
        agregarProductoAlCarritoFav={() => {}}
        removeItemFromCartFav={() => {}}
        cart={["Test"]}
        favProducts={[]}
        laptop={{ id: 1, name: "Test", price: "6.7" }}
      ></Laptop>
    </Router>
  );

  tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});
