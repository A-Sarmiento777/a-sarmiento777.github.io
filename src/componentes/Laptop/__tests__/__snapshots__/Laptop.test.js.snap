// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Laptop Card test 1`] = `
<div
  className="card mr-4 mb-4 p-3 shadow p-3 mb-5 bg-white rounded "
  style={
    Object {
      "minWidth": "30px",
      "width": "200px",
    }
  }
>
  <div
    className="h6"
  >
    Test
  </div>
  <div
    className="h6"
  >
     
    <small>
       Pris: 
       
    </small>
  </div>
  <div
    className="mx-auto"
  >
    <img
      alt="logos"
      className="img-fluid py-2"
      src="/Assets/laptops/Test.jpg"
      width="150"
    />
  </div>
  <div>
    <button
      className="bg-light border-0"
      onClick={[Function]}
    >
      <img
        alt="logo"
        src="/Assets/fav.png"
        width="20"
      />
    </button>
    <button
      className="bg-warning px-1 rounded ml-2"
      onClick={[Function]}
    >
      LÃ¤gg till
    </button>
  </div>
</div>
`;

exports[`Laptop Card test 2`] = `
<div
  className="card mr-4 mb-4 p-3 shadow p-3 mb-5 bg-white rounded "
  style={
    Object {
      "minWidth": "30px",
      "width": "200px",
    }
  }
>
  <div
    className="h6"
  >
    Test
  </div>
  <div
    className="h6"
  >
     
    <small>
       Pris: 
       
    </small>
  </div>
  <div
    className="mx-auto"
  >
    <img
      alt="logos"
      className="img-fluid py-2"
      src="/Assets/laptops/Test.jpg"
      width="150"
    />
  </div>
  <div>
    <button
      className="bg-light border-0"
      onClick={[Function]}
    >
      <img
        alt="logo"
        src="/Assets/fav.png"
        width="20"
      />
    </button>
    <a
      href="/checkout"
      onClick={[Function]}
    >
      <button
        className="bg-success px-1 rounded ml-2"
      >
        Varukorgen
      </button>
    </a>
  </div>
</div>
`;
